### PR TITLE
Add fake rest provider

### PR DESCRIPTION
**Pull request by jonathan-marmelab**
## Problem

We want to support both supabase and fake rest as data providers

## Solution

Add support for fake rest data provider that matches supabase data provider methods
 
## How To Test

Use fake rest data providers and auth providers in CRM component

## Additional Checks

- [x] The **documentation** is up to date

### CODE REVIEW

File: src/providers/fakerest/dataProvider.ts [hunk 1]
```diff
@@ -20,10 +20,11 @@ import { getActivityLog } from '../commons/activity';
 import { getCompanyAvatar } from '../commons/getCompanyAvatar';
 import { getContactAvatar } from '../commons/getContactAvatar';
 import { CrmDataProvider } from '../types';
+import { authProvider, USER_STORAGE_KEY } from './authProvider';
 import generateData from './dataGenerator';
 import { withSupabaseFilterAdapter } from './internal/supabaseAdapter';
 
-const baseDataProvider = fakeRestDataProvider(generateData(), true, 300);
+const baseDataProvider = fakeRestDataProvider(generateData(), false, 300);
```
Comment (slax57) [inline src/providers/fakerest/dataProvider.ts:27]: question: why that change?

File: src/providers/fakerest/dataProvider.ts [hunk 2]
```diff
@@ -241,6 +242,16 @@ export const dataProvider = withLifecycleCallbacks(
...
+            afterSave: async data => {
+                const currentUser = await authProvider.getIdentity?.();
+                if (currentUser?.id === data.id) {
+                    localStorage.setItem(
+                        USER_STORAGE_KEY,
+                        JSON.stringify(data)
+                    );
+                }
+                return data;
+            },
```
Comment (slax57) [inline src/providers/fakerest/dataProvider.ts:254]: suggestion: add a little comment to explain why this is necessary? (it certainly is not obvious for me at least)

File: src/providers/fakerest/dataProvider.ts (hunk 1 above)
Comment (jonathan-marmelab) [reply src/providers/fakerest/dataProvider.ts:28]: Too much logs in the console, debug is awful with this feature to true

**Review by slax57 — APPROVED** [review]
Couldn't find anything wrong with the code or by testing, nice work!!

**Pull request merged**